src/elisp/treemacs-themes.el

```diff
@@ -38,6 +38,9 @@
 
 (defvar treemacs--themes nil "List of all known themes.")
 
+(defvar treemacs-modes '(treemacs-mode dired-mode)
```

Comment:
Let's call this `treemacs-icon-modes` so it's clear which modes this is about.

The commit message should also be start with `[Icons]`.